src/internal/adapter/THEOplayerAdapter.ts

```diff
@@ -225,9 +225,22 @@ export class THEOplayerAdapter extends DefaultEventDispatcher<PlayerEventMap> im
     return this._adsAdapter;
   }
 
+  private applyAutoplay() {
+    if (this._state.autoplay) {
+      this.play();
+    } else {
+      this.pause();
+    }
+  }
+
   set autoplay(autoplay: boolean) {
     this._state.autoplay = autoplay;
-    NativeModules.PlayerModule.setPaused(this._view.nativeHandle, !autoplay);
+
+    // Apply autoplay in case a source was already set.
+    // If autoplay is changed before setting a source, `onSourceChange` applies autoplay.
+    if (this.hasValidSource()) {
```

Comment:
Maybe just do the applyAutoplay() here. And move the hasValidSource() check inside the applyAutoplay() method making it more robust every time it is called (i.e. also onSourcechange)